Fix websocket pong

a1rwulf opened the pull request:
## Description
After updating my web remote to latest packages and node, I was surprised by websocket disconnects.
Debugging showed after every ping I did, my connection was aborted with following error:
1007 Illegal UTF-8 Sequence

Furthermore while investigating the issue I found that our implementation violates the RFC as specified here:
https://datatracker.ietf.org/doc/html/rfc6455#section-5.5.3

Prior to this PR, our pong always contained an empty body.

## How has this been tested?
Run this using current node 18 LTS.
Run `npm i ws` to install the websocket client library.

```
const WS = require('ws');

const ws = new WS('ws://localhost:9090/jsonrpc');

ws.on('open', () => {
    console.log('connected');
});

ws.on('message', (data) => {
    console.log('message received');
});

ws.on('close', () => {
    console.log('closed');
});

ws.on('error', (err) => {
    console.log('error', err);
});

ws.on('pong', (appdata) => {
    console.log(`got pong: ${appdata}`);
});

const keepAliveId = setInterval(() => { ws.ping('test'); }, 1500);
```

## What is the effect on users?
Properly working remotes.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

a1rwulf committed:
Properly add application data to the websocket pong

The RFC states that pong should reply with the same application
data that is received in the ping.
Our code however threw away the application data instead.
We need to properly set application data length so that it actually
gets appended to the send buffer.

a1rwulf commented:
Also note that if you switch back to ws 7.2.1 for example, you'll receive a message instead of a pong, with some non UTF characters, because ws 7.2.1 does not hard fail when receiving such invalid data.

lrusak commented on xbmc/network/TCPServer.cpp:
```diff
@@ -719,7 +719,8 @@ void CTCPServer::CWebSocketClient::PushBuffer(CTCPServer *host, const char *buff
         for (unsigned int index = 0; index < frames.size(); index++)
-          Send(frames.at(index)->GetFrameData(), (unsigned int)frames.at(index)->GetFrameLength());
+          CTCPClient::Send(frames.at(index)->GetFrameData(),
+                           (unsigned int)frames.at(index)->GetFrameLength());
```
Can you please update the cast to c++ style?

lrusak commented on xbmc/network/websocket/WebSocket.h:
```diff
@@ -121,7 +121,7 @@ class CWebSocket
   virtual const CWebSocketMessage* Send(WebSocketFrameOpcode opcode, const char* data = NULL, uint32_t length = 0);
   virtual const CWebSocketFrame* Ping(const char* data = NULL) const = 0;
-  virtual const CWebSocketFrame* Pong(const char* data = NULL) const = 0;
+  virtual const CWebSocketFrame* Pong(const char* data = NULL, uint32_t length = 0) const = 0;
```
Is the default value of 0 really needed?

a1rwulf replied:
It is, if we keep the data optional.
There is only one implementation, so it's trivial to remove them both.
It's a matter of taste, application data is optional for a pong, so semantically I think the defaults would be fine.
Changed it anyway, I'll keep whatever is preferred.

a1rwulf commented:
Changed the wrong line, after we decided wether to keep default params for this or not, I'll squash and make sure the non-modified code stays untouched.

a1rwulf committed:
Fix websocket pong and probably other messages

When "handle" gets called it already creates a perfectly
fine "answer"-frame.
Calling the Send method of CWebSocketClient will wrap it into
another TextFrame which may result in an  1007 Illegal UTF-8 Sequence
or possibly other invalid websocket data frames.
We need to call Send of CTCPClient instead which will just transfer
the frame without further modification.

a1rwulf commented:
Alright, should be good after jenkins is fine with it.

a1rwulf commented:
Jenkins build this please
